lib/modules/varnish/poll.c

```diff
@@ -65,27 +65,38 @@ static const struct {
 size_t varnish_fetch_metrics(varnish_module_t *mod, const char *output) {
   size_t n_metrics = 0;
   char *found;
+  char *output_start;
   char metric_name[256];
   struct timeval clock;
 
   gettimeofday(&clock, NULL);
+  output_start = output;
```

Comment:
reuse the "found" variable instead (reset it above the while to output and do `found += strlen(metrics[i].varnish_name)` instead of `output = &found[strlen(metrics[i].varnish_name)]`
Also a for loop would be good, see below